kitsune/wiki/models.py

```diff
@@ -751,19 +752,28 @@ def is_restricted(self):
         return self.restrict_to_groups.exists()
 
     def is_unrestricted_for(self, user):
-        """Is the given user unrestricted in their visibility of this document?"""
+        """
+        Is the given user unrestricted in their visibility of this document?
+        """
         if self.parent:
             # Translations follow their parent's restrictions.
             return self.parent.is_unrestricted_for(user)
 
-        return bool(
+        return (
             user.is_staff
             or user.is_superuser
             or (not self.is_restricted)
-            or (
-                user.is_authenticated
-                and (set(self.restrict_to_groups.all()) & set(user.groups.all()))
-            )
+            or (user.id in self.restrict_to_user_ids)
+        )
+
+    @cached_property
+    def restrict_to_user_ids(self):
+        """
+        Returns a set of user ids comprising all members of the groups specified in
+        "restrict_to_groups".
+        """
+        return set(
```

Comment:
I would move this query into the `is_unrestricted_for` method. We are adding a property that's not used anywhere else. Let's move it in the above method and if we can extract it if we ever need it in another place.